knowledge_repo/app/index.py

```diff
@@ -124,7 +124,34 @@ def index_up_to_date():
             return False
     return True
 
+@ErrorLog.logged
+def update_index_for_post(kp,path):
+    import time
+    is_index_master = acquire_index_lock()
+    
+    try:
+        IndexMetadata.set('lock', 'index', LOCKED)
+        db_session.commit()
+        if not kp.is_valid():
+            logger.warning(u'New post at "{}" is corrupt.'.format(kp.path))
+            return
+        querypath = path+'.kp'
+        post = (db_session.query(Post).filter(Post.path==querypath).first())
+        if not post:
+            logger.info(u'creating new post from path {}'.format(kp.path))
+            post = Post()
```

Comment:
@abhirathb overwriting post variable? Is that not required?

Reply:
yep. The idea being that either the post will exist already in the DB, or a new object must be created for it. After the if block, I update the metadata on the post, agnostic of whether it was newly created or fetched from the DB